### DISCUSSION TITLE
LoRa Communication (Non-Arduino) for Transmitting and Receiving Data in VS Code + STM32 Official Extension or STM32CubeIDE

### DISCUSSION THREAD

**Discussion by mkumar2789** [jgromes/RadioLib#1337] [Q&A]
Hi,
I require assistance in developing a straightforward code for transmitting and receiving a single byte using the LoRa SX1278 (Ra-02) module with STM32F103. Specifically, I need guidance on selecting the appropriate files (.h and .c/.cpp) from the RadioLib library for my application and configuring my project. I am seeking step-by-step instructions to accomplish this task.
Any assistance in this matter would be greatly appreciated.

**Comment by jgromes** [comment]
IMO the easiest way is to use CMake (which should be supported even by CubeIDE if I'm not mistaken), then just add the entire RadioLib directory as subdirectory (see an example [here](https://github.com/jgromes/RadioLib/blob/master/examples/NonArduino/Raspberry/CMakeLists.txt)) and then simply `#include <RadioLib.h>`.

> the appropriate files (.h and .c/.cpp)

All of them, `src/*.h` and `src/*.h`. I don't know your exact setup - I don't use CubeIDE and only use VS code as text editor with integrated git support, then build everything using scripts as the NonArduino examples show.

**Comment by HeadBoffin** [comment]
> I am seeking step-by-step instructions to accomplish this task.

What you are asking for would be a lot of work & be rather specific to your needs. Particularly using the STM32 environment in CVSC or CubeIDE.

You may find more resources available if you change your MCU or use STMduino in VSC with RadioLib.
